lang/ruby/lib/avro/schema.rb

```diff
@@ -467,6 +475,11 @@ def to_avro(names=nil)
         hsh = super
         hsh.size == 1 ? type : hsh
       end
+
+      def match_schema?(schema)
+        return type_sym == schema.type_sym
```

Comment:
Is this essentially just making sure the writer schema is a record as well?

Reply:
This is actually just for the primitive, non-named types to ensure that they are the same type.

Primarily it is used from this modified line https://github.com/apache/avro/pull/1082/files/7ea99640fd96204dcb217133da6c3c6394ef043c#diff-5fa6692791f1794cd9064f4288aff1c2447925b993f77196d09b9bb6a05c910fR49, since that call only applies to primitive types only. 

It's somewhat redundant right now since the call is nested within a `w_type == r_type` condition. But I was trying to move the compatibility checked into a more object-oriented direction.

It is also called from as `super` from the `BytesSchema#match_schema?`.